plugins/pencilblue/controllers/actions/admin/users/send_password_reset.js

```diff
@@ -79,10 +79,14 @@ module.exports = function(pb) {
                 }
 
                 if(!passwordReset) {
-                    passwordReset = pb.DocumentCreator.create('password_reset', {user_id: user[pb.DAO.getIdField()].toString()});
+                    passwordReset = {
+                        userId: user[pb.DAO.getIdField()].toString(),
+                        object_type: 'password_reset'
+                    };
                 }
 
-                passwordReset.verification_code = util.uniqueId();
+                passwordReset.verificationCode = util.uniqueId();
+                passwordReset.site = self.context.site;
```

Comment:
Instead of manually setting the site, you can change line 52 such that: ```this.dao = new SiteQueryService(self.getServiceContext());```.  This will ensure that the site is automatically injected when the object is saved.